core/storage/user/gae_models.py

```diff
@@ -288,6 +324,18 @@ def get_deletion_policy():
         """
         return base_models.DELETION_POLICY.DELETE
 
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether ExpUserLastPlaythroughModels exist for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether the models for user_id exists.
+        """
+        return cls.get_all().filter(cls.user_id == user_id).get() is not None
```

Comment:
Here and below, shouldn't it be cls.query(cls.user_id == user_id).get() ? https://cloud.google.com/appengine/docs/standard/python/ndb/queries#filter_by_prop

Reply:
`get_all()` returns `Query` object but with models that are not marked as deleted. The question is if we want to detect models that are marked as deleted.

Reply:
Ah, got it. I think the answer is yes -- we don't just want to mark the models as deleted, we want to actually delete the models. So we should surface all models that need to be hard-deleted.

Reply:
Done, also changed the `.get(user_id)` to `.get_by_id(user_id)` for the same reason (to retrive models marked as deleted).